sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala

```diff
@@ -820,8 +820,12 @@ class DateExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     test("2015-07-24 00:00:00", null, null)
     test(null, null, null)
     // Test escaping of timezone
-    GenerateUnsafeProjection.generate(
-      ToUTCTimestamp(Literal(Timestamp.valueOf("2015-07-24 00:00:00")), Literal("\"quote")) :: Nil)
+    val msg = intercept[java.time.DateTimeException] {
```

Comment:
Does this indicate a behavior change? I wasn't sure why this should change.

Reply:
The time zone id `"\"quote"` is invalid. Previously it was just ignored silently.

Reply:
I see, so that was probably a bug. It wouldn't hurt to note that with `release-notes` tag and `Docs text` in the JIRA, as it might cause invalid data to now (correctly) fail to parse.

Reply:
PS this is fine to leave in as a test, but then this is no longer testing what it seems to intend to, which is an 'escaped' timezone. But then, I'm not sure what that would be? nothing in valid timezone strings needs escaping that I can think of.

If so just remove the comment about 'test escaping of timezone'.

Reply:
One more question: from my reading of the code, previously, an invalid timezone would result in using GMT, from reading TimeZone. If that's right I'll add it as release notes.

Reply:
I updated the SQL migration guide